sdk/python/feast/client.py

```diff
@@ -400,27 +428,105 @@ def get_batch_features(
         return Job(response.job, self._serving_service_stub)
 
     def _validate_entity_rows_for_batch_retrieval(
-        self, entity_rows, feature_sets_request
+        self, entity_rows: pd.DataFrame, feature_sets_request
     ):
         """
-        Validate whether an entity_row dataframe contains the correct
-        information for batch retrieval
+        Validate whether an entity_row DataFrame contains the correct
+        information for batch retrieval.
+
+        Datetime column must be present in the DataFrame.
 
         Args:
-            entity_rows: Pandas dataframe containing entities and datetime
-                column. Each entity in a feature set must be present as a
-                column in this dataframe.
-            feature_sets_request: Feature sets that will be requested
+            entity_rows (pd.DataFrame):
+                Pandas DataFrame containing entities and datetime column. Each
+                entity in a feature set must be present as a column in this
+                DataFrame.
+
+            feature_sets_request:
+                Feature sets that will be requested.
         """
 
+        self._validate_columns(
+            columns=entity_rows.columns,
+            feature_sets_request=feature_sets_request,
+            datetime_field="datetime"
+        )
+
+    def _validate_avro_for_batch_retrieval(
+            self, source: str, feature_sets_request
+    ):
+        """
+        Validate whether an Avro source file contains the correct information
+        for batch retrieval. Only gs:// and local files (file://) uri schemes
+        are allowed.
+
+        Avro file must have a column named "event_timestamp".
+
+        No checks will be done if a GCS path is provided.
+
+        Args:
+            source (str):
+                File path to Avro.
+
+            feature_sets_request:
+                Feature sets that will be requested.
+        """
+        p = urlparse(source)
+
+        if p.scheme == "gs":
+            # GCS path provided (Risk is delegated to user)
+            # No validation if GCS path is provided
+            return
+        elif p.scheme == "file" or not p.scheme:
+            # Local file (file://) provided
+            file_path = os.path.abspath(os.path.join(p.netloc, p.path))
+        else:
+            raise Exception(f"Unsupported uri scheme provided {p.scheme}, only "
+                            f"local files (file://), and gs:// schemes are "
+                            f"allowed")
+
+        with open(file_path, "rb") as f:
+            reader = fastavro.reader(f)
+            schema = json.loads(reader.metadata["avro.schema"])
+            columns = [x["name"] for x in schema["fields"]]
+            self._validate_columns(
+                columns=columns,
+                feature_sets_request=feature_sets_request,
+                datetime_field="event_timestamp"
+            )
+
+    def _validate_columns(
```

Comment:
should we rename this to something entity_rows specific?

Reply:
> @voonhous this looks pretty good, thanks!
> 
> Would you mind updating the tests as well? I think it's important to ensure that the code actually works.

Sure thing! Working on it now!

Reply:
> should we rename this to something entity_rows specific?

I'll rename this to `_validate_entity_columns`.

In summary, `_validate_entity_rows_for_batch_retrieval` will be renamed to `_validate_dataframe_for_batch_retrieval`.


In summary:

`_validate_dataframe_for_batch_retrieval` and `_validate_avro_for_batch_retrieval` uses `_validate_entity_columns` to check if the required columns matches that specified in the feature_sets_request.

Reply:
Ok